server/scsynth/SC_PortAudio.cpp

```diff
@@ -341,9 +338,25 @@ bool SC_PortAudioDriver::DriverSetup(int* outNumSamples, double* outSampleRate)
 	mDeviceInOut[0] = paNoDevice;
 	mDeviceInOut[1] = paNoDevice;
 	if (mWorld->hw->mInDeviceName)
-		GetPaDeviceFromName(mWorld->hw->mInDeviceName, mDeviceInOut);
+		GetPaDeviceFromName(mWorld->hw->mInDeviceName, &mDeviceInOut[0], true);
+	if (mWorld->hw->mOutDeviceName)
+		GetPaDeviceFromName(mWorld->hw->mOutDeviceName, &mDeviceInOut[1], false);
+
+	#ifdef _WIN32
```

Comment:
in this codebase we place all preprocessor directives at the far left and indent the part after the hash if they are nested. also it's nice to leave a comment at the `endif` (and optionally `else`) noting the condition:

```cpp
#ifdef _WIN32
...
#    define X Y /* nested directive */
...
#else
...
#endif // _WIN32

Reply:
see for instance lines 367-382 of scsynth_main.cpp (https://github.com/supercollider/supercollider/blob/de6a0c42bb6246344bf5b7a18c11f98a5bd1b987/server/scsynth/scsynth_main.cpp#L367-L382)